pkg/auth/authorizer/abac/abac.go

```diff
@@ -100,13 +101,20 @@ func NewFromFile(path string) (policyList, error) {
 func (p policy) matches(a authorizer.Attributes) bool {
 	if p.subjectMatches(a) {
 		if p.Readonly == false || (p.Readonly == a.IsReadOnly()) {
-			if p.Resource == "" || (p.Resource == a.GetResource()) {
+			switch {
+			case p.NonResourcePath != "":
```

Comment:
if the incoming attributes indicate a non-resource, we always want to choose this branch, right? Shouldn't the check be this:
```
case len(a.GetNonResourcePath()) > 0:
```